wallet/how-to/use-non-evm-networks/starknet/manage-starknet-accounts.md

```diff
@@ -0,0 +1,238 @@
+
+```typescript
+import { useStarknet, useConnectors } from '@starknet-react/core';
+import { useState, useEffect } from 'react';
+
+function AccountDisplay() {
+  const { account } = useStarknet();
+  const { available, connect, disconnect } = useConnectors();
+  const [accountAddress, setAccountAddress] = useState<string | undefined>();
+
+  useEffect(() => {
+    setAccountAddress(account?.address);
+  }, [account]);
+
+  return (
+    <div>
+      {accountAddress ? (
+        <p>Connected Account: {accountAddress}</p>
+      ) : (
+        <p>No account connected</p>
+      )}
+      {available.map((connector) => (
+        <button key={connector.id()} onClick={() => connect(connector)}>
+          Connect {connector.name()}
+        </button>
+      ))}
+      {account && (
+        <button onClick={() => disconnect()}>Disconnect</button>
+      )}
+    </div>
+  );
+}
+```
+
+This component will display the connected account address if available, and provide buttons to connect or disconnect accounts.
+
+## 2. Switching Between Multiple Starknet Accounts
+
+Starknet doesn't natively support account switching like Ethereum. Instead, users typically have to disconnect and reconnect with a different account. However, you can implement a user-friendly interface to manage this process:
+
+```typescript
+import { useStarknet, useConnectors } from '@starknet-react/core';
+import { useState, useEffect } from 'react';
+
+function AccountSwitcher() {
+  const { account } = useStarknet();
+  const { available, connect, disconnect } = useConnectors();
+  const [connectedAccounts, setConnectedAccounts] = useState<string[]>([]);
+
+  useEffect(() => {
+    if (account?.address && !connectedAccounts.includes(account.address)) {
+      setConnectedAccounts([...connectedAccounts, account.address]);
+    }
+  }, [account]);
+
+  const switchAccount = async (connector: any) => {
+    await disconnect();
+    await connect(connector);
+  };
+
+  return (
+    <div>
+      <h2>Connected Accounts:</h2>
+      <ul>
+        {connectedAccounts.map((address) => (
+          <li key={address}>{address}</li>
+        ))}
+      </ul>
+      <h2>Available Connectors:</h2>
+      {available.map((connector) => (
+        <button key={connector.id()} onClick={() => switchAccount(connector)}>
+          Switch to {connector.name()}
+        </button>
+      ))}
+    </div>
+  );
+}
+```
+
+This component keeps track of all connected accounts and allows switching between available connectors.
+
+## 3. Account Creation Process
+
+Account creation in Starknet is typically handled by the wallet provider (e.g., ArgentX, Braavos). As a dApp developer, you don't create accounts directly. Instead, you guide users to create an account with their preferred wallet provider. Here's how you might implement a guide for users:
+
+```typescript
```

Comment:
Note that the `connect` method of `get-starknet` 

```
import { connect } from "get-starknet"

return <button onClick={() => connect()}>Connect wallet</button>
```

When clicking on the connect wallet button above the users will see the list of compatible wallet, will be able to click to install one or use one already installed. 

This interface can be modified if needed, more information is available on get-starknet GitHub